block-producer/src/server/api.rs

```diff
@@ -32,15 +33,31 @@ impl<T> api_server::Api for BlockProducerApi<T>
 where
     T: TransactionQueue,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip_all, err, fields(COMPONENT = crate::COMPONENT))]
     async fn submit_proven_transaction(
```

Comment:
> Also, I'm thinking we should override the name and use block_producer::submit_proven_transaction. It is a bit redundant with the COMPONENT filed, but would make the pretty output more readable.

I will change it for all API handlers, thanks.

> Lastly, in case there is an error, will this use Display or Debug trait to serialize the error? (we should probably use Display and implement good error messages on errors in subsequent PRs).

We use `Debug`, but will be able to change to `Display` if implement pretty `Display` for errors.